source/SIL.AppBuilder.Portal.Frontend/src/ui/routes/projects/show/products/index.tsx

```diff
@@ -29,21 +29,34 @@ interface IOwnProps {
   isEmptyWorkflowProjectUrl: boolean;
 }
 
+interface IPendingUpdates {
+  [itemId: string]: boolean;
+}
+
 type IProps =
   & IOwnProps
   & i18nProps
   & IDataActionsProps;
 
 class Products extends React.Component<IProps> {
+  pendingUpdates: IPendingUpdates;
 
-  onSelectionChange = async (item: ProductDefinitionResource) => {
+  constructor(props: IProps) {
+    super(props);
+    this.pendingUpdates = {};
+  }
 
-    const { t, updateProduct } = this.props;
-    try {
-      await updateProduct(item);
-      toast.success(t('updated'));
-    } catch (e) {
-      toast.error(e.message);
+  onSelectionChange = async (item: ProductDefinitionResource) => {
+    if (!this.pendingUpdates[item.id]){
+      this.pendingUpdates[item.id] = true;
```

Comment:
nice use of of state outside of the recommended state management! :)

using setState triggers re-renders, so this is perfect.

Reply:
yep.